src/windows-emulator/windows_emulator.cpp

```diff
@@ -448,9 +448,24 @@ void windows_emulator::setup_hooks()
     });
 
     this->emu().hook_instruction(x64_hookable_instructions::rdtsc, [&] {
-        const auto instructions = this->executed_instructions_;
-        this->emu().reg(x64_register::rax, instructions & 0xFFFFFFFF);
-        this->emu().reg(x64_register::rdx, (instructions >> 32) & 0xFFFFFFFF);
+        static bool first_rdtsc = 0;
+        static uint_fast64_t rdtsc_prev = 0;
+
+        uint64_t rdtsc_real = __rdtsc();
```

Comment:
this doesn't work on non intel devices.
question is, what exactly does rdtsc return? afaik the cpu ticks? which probably somewhat correlates to the amount of executed instructions?

also, this makes the emulation non-deterministic. i think it might make sense to either extend the clock interface, or to precisely reimplement rdtsc. 

question is, what happens if you just remove the hook and use the default implementation of the underlying emulator?

Reply:
> this doesn't work on non intel devices. question is, what exactly does rdtsc return? afaik the cpu ticks? which probably somewhat correlates to the amount of executed instructions?
> 
> also, this makes the emulation non-deterministic. i think it might make sense to either extend the clock interface, or to precisely reimplement rdtsc.
> 
> question is, what happens if you just remove the hook and use the default implementation of the underlying emulator?

Yeah i just realized the project builds for non x86 systems. This does build on non intel CPUs though considering im using AMD myself